onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -52,7 +53,191 @@ static void BuildFusedKernelDef(KernelDefBuilder& builder, const IndexedSubGraph
+        continue;
+      }
+
+      // if already transformed then change the domain to kMSNHWCDomain this way the EP
+      // knows this op is in the expected format.
+      if (node->GetAttributeIntDefault("channels_last", 0) == 1) {
+        onnx_layout_transformation::SwapNodeOpTypeAndDomain(*api_graph, *node, node->OpType(), kMSNHWCDomain);
+        // Changing the domain for the node requires creating a new node and replacing the old one
+        // therefore set the modified flag.
+        modified = true;
+        continue;
+      }
+
+      // Skip if unknown rank
+      auto shape = NodeFromApiNode(*node).InputDefs()[0]->Shape();
+      if (shape == nullptr) {
+        continue;
+      }
+
+      // Convert to channels last
+      size_t rank = shape->dim_size();
+
+      bool has_channel_last_attr = node->GetAttributeInt("channels_last") != nullopt ? true : false;
+      if (has_channel_last_attr) {
+        node->SetAttributeInt("channels_last", 1);
+      }
+
+      auto input_perm = onnx_layout_transformation::ChannelFirstToLastPerm(rank);
+      auto output_perm = onnx_layout_transformation::ChannelLastToFirstPerm(rank);
+
+      // Except for resize and convolution ops, all the other layout sensitive ops only require layout transformation 
+      // for 0th input and output. For resize, add the other relevant inputs which need conversion. For Conv - layout 
+      // transformer only converts layout for 0th input, weights should be handled by every EP.
+      if (node->OpType() == "Resize") {
+        // Older versions of resize have a bug where ROI and Scales cannot be made empty inputs. To handle this case, 
+        // we need to jump a few extra hoops to make sure its inputs are correctly handled. Current code skips 
+        // layout conversion for ROI because it needs special handling as ROI size is 2*rank. 
+        // Enable passing in ROI for layout conversion when an EP which supports ROI starts using layout transformer.
+        // NNAPI which currently uses layout transformer does not support it.
+        std::vector<const std::vector<int64_t>*> input_perms{&input_perm, nullptr};
+        for (size_t i = 2; i < node->Inputs().size(); i++) {
+          auto constant = api_graph->GetConstant(node->Inputs()[i]);
+          if (constant != nullptr && constant->Data().size() > 0) {
+            input_perms.push_back(&input_perm);
+          } else {
+            input_perms.push_back(nullptr);
+          }
+        }
+        onnx_layout_transformation::WrapTransposesAroundNode(*api_graph, *node, input_perms, {&output_perm});
+      } else {
+        onnx_layout_transformation::WrapTransposesAroundNode(*api_graph, *node, {&input_perm}, {&output_perm});
+      }
+
+      onnx_layout_transformation::SwapNodeOpTypeAndDomain(*api_graph, *node, node->OpType(), kMSNHWCDomain);
+      modified = true;
+    }
+  }
+
+  if (modified) {
+    modified = onnx_layout_transformation::Optimize(*api_graph,
```

Comment:
>modified [](http://example.com/codeflow?start=4&length=8)

this assignment to modified is not necessary. effectively, it is `modified = ... || true` as modified is true within this block